src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -927,7 +928,7 @@ private class DeclarationAnalysisData
         internal AnalyzerDriver(ImmutableArray<DiagnosticAnalyzer> analyzers, Func<SyntaxNode, TLanguageKindEnum> getKind, AnalyzerManager analyzerManager) : base(analyzers, analyzerManager)
         {
             _getKind = getKind;
-            _declarationAnalysisDataCache = new ConditionalWeakTable<SyntaxReference, DeclarationAnalysisData>();
+            _declarationAnalysisDataCache = new ConditionalWeakTable<Compilation, Dictionary<Location, DeclarationAnalysisData>>();
```

Comment:
:thought_balloon: This may help us in StyleCopAnalyzers. `ConditionalWeakTable` has a lock inside of `TryGetValue` which is causing heavy contention problems in my 28-core code analysis test runs.